src/types/entities/player.ts

```diff
@@ -22,6 +22,7 @@ export class Player {
 	private line: number // where this player is in the players list TODO: -1 for not visible
 	private _isPlaying = false
 	private lastBlock: BlockDetails | undefined // block details of last interaction
+	private lastAction: string | undefined // tracks last action which cannot be derived from blockNumber
```

Comment:
I'd most probably suggest that given that the actions are limited and defined, that an enum should be used for this as opposed to a string to more strictly enforce typing.